pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -319,8 +324,10 @@ func (hsm *hostStateMachine) handleProvisioned(info *reconcileInfo) actionResult
 func (hsm *hostStateMachine) handlePowerManagementError(info *reconcileInfo) actionResult {
 	switch {
 	case hsm.Host.Spec.ExternallyProvisioned:
+		info.host.ResetErrorCount()
 		hsm.NextState = metal3v1alpha1.StateExternallyProvisioned
 	case hsm.Host.WasProvisioned():
+		info.host.ResetErrorCount()
```

Comment:
This is not the time to reset the error count. If you do it here, the sequence of events will be something like:

* Host is in `StateProvisioned`
* Power management fails, increment error count to 1, return `actionFailed` and change state to `StatePowerManagementError`
* Wait 1-2s (2^(error count) * [0.5,1.0])
* Reset error count to 0, return to `StateProvisioned`
* Repeat from the top

We need to reset the error count only when we successfully complete a power management operation. Probably doing this in `ClearError()` would be close enough for our purposes (there are two different types of errors in `actionManageSteadyState()` and avoiding *either* of them clears errors, which is why I suggested maintaining different counts for different error types, but I now think that is overkill). Semantically it makes sense too (iff there is no current error, then the error count is 0).

Plus, as Doug said, these error states are going away in #388 Real Soon Now&trade;, so we should avoid adding new code in here, since it will have to go somewhere else anyway.